src/ucp/rndv/rndv_put.c

```diff
@@ -345,16 +348,23 @@ static UCS_F_ALWAYS_INLINE ucs_status_t ucp_proto_rndv_put_zcopy_send_func(
         ucp_datatype_iter_t *next_iter)
 {
     const ucp_proto_rndv_put_priv_t *rpriv = req->send.proto_config->priv;
+    ucp_lane_index_t lane_shift;
     size_t max_payload;
     uct_iov_t iov;
+    ucs_status_t status;
 
-    max_payload = ucp_proto_rndv_bulk_max_payload(req, &rpriv->bulk, lpriv);
+    max_payload = ucp_proto_rndv_bulk_max_payload_align(req, &rpriv->bulk,
+                                                        lpriv, &lane_shift);
     ucp_datatype_iter_next_iov(&req->send.state.dt_iter, max_payload,
                                lpriv->super.md_index,
                                UCS_BIT(UCP_DATATYPE_CONTIG), next_iter, &iov,
                                1);
-    return ucp_proto_rndv_put_common_send(req, lpriv, &iov,
-                                          &req->send.state.uct_comp);
+    status = ucp_proto_rndv_put_common_send(req, lpriv, &iov,
```

Comment:
the send can be unsuccessful (i.e. ` status == NO_RESOURCE`), then no need to change the lane

Reply:
remove 'status' helper variable and just return the result from ucp_proto_rndv_put_common_send